### PR TITLE
Document that `CallResponse.returnDependencies` is augmented by the engine

### PR DESCRIPTION
**Pull request by iwahbe**
This is where the engine augments `returnDependencies` with `Output` dependencies:

https://github.com/pulumi/pulumi/blob/cbfea508e7b2a4f1dd0985cefbeda3a6e891f7f9/pkg/resource/deploy/source_eval.go#L1140-L1142

This is where we test that this behavior works as intended:

https://github.com/pulumi/pulumi/blob/44a95c434a48a8b8153fcc419d81b5a22a961350/pkg/engine/lifecycletest/pulumi_test.go#L5041-L5042

This was motivated by https://github.com/pulumi/pulumi-go-provider/pull/263#discussion_r1733224233.

### CODE REVIEW

**Comment by Frassle** [comment]
Where would you expect that to be documented?

**Comment by iwahbe** [comment]
> Where would you expect that to be documented?

In the `provider.proto` file, where I put the documentation.

**Review by tgummerer — APPROVED** [review]
Maybe the title should be "Document that `CallResponse.returnDependencies` is augmented by the engine", rather than it being unnecessary?  (It ends up in the git log, so it would be nice to be accurate)

File: proto/pulumi/provider.proto [hunk 1]
```diff
@@ -221,9 +221,14 @@ message CallResponse {
...
-    google.protobuf.Struct return = 1;                      // the returned values, if call was successful.
-    map<string, ReturnDependencies> returnDependencies = 2; // a map from return value keys to the dependencies of the return value.
-    repeated CheckFailure failures = 3;                     // the failures if any arguments didn't pass verification.
+    google.protobuf.Struct return = 1;  // the returned values, if call was successful.
+    repeated CheckFailure failures = 3; // the failures if any arguments didn't pass verification.
+
+    // a map from return value keys to the dependencies of the return value.
+    //
+    // returnDependencies will be augmented by the set of dependencies specified in return
+    // via serialized github.com/pulumi/pulumi/sdk/v3/go/common/resource.Output.
```
Comment (Frassle) [inline proto/pulumi/provider.proto:230]: This shouldn't refer to Go code

**Review by Frassle — CHANGES_REQUESTED** [review]

File: proto/pulumi/provider.proto (hunk 1 above)
Comment (iwahbe) [reply proto/pulumi/provider.proto:230]: How would you recommend I refer to outputs? I don't believe that this is present at the gRPC layer.

Comment (Frassle) [reply proto/pulumi/provider.proto:230]: Just say "output property value"?

**Review by Frassle — APPROVED** [review]

**Pull request merged**